raphtory/src/python/filter/property_filter_builders.rs

```diff
@@ -231,74 +243,165 @@ impl PyPropertyFilterOps {
             .fuzzy_search(prop_value, levenshtein_distance, prefix_match)
     }
 
-    fn len(&self) -> PyPropertyFilterOps {
+    pub fn any(&self) -> PyResult<PyPropertyFilterOps> {
+        self.qual.any()
+    }
+
+    pub fn all(&self) -> PyResult<PyPropertyFilterOps> {
+        self.qual.all()
+    }
+
+    fn len(&self) -> PyResult<PyPropertyFilterOps> {
         self.agg.len()
     }
 
-    fn sum(&self) -> PyPropertyFilterOps {
+    fn sum(&self) -> PyResult<PyPropertyFilterOps> {
         self.agg.sum()
     }
 
-    fn avg(&self) -> PyPropertyFilterOps {
+    fn avg(&self) -> PyResult<PyPropertyFilterOps> {
         self.agg.avg()
     }
 
-    fn min(&self) -> PyPropertyFilterOps {
+    fn min(&self) -> PyResult<PyPropertyFilterOps> {
         self.agg.min()
     }
 
-    fn max(&self) -> PyPropertyFilterOps {
+    fn max(&self) -> PyResult<PyPropertyFilterOps> {
         self.agg.max()
     }
 }
 
 pub trait DynListAggOps: Send + Sync {
-    fn len(&self) -> PyPropertyFilterOps;
-    fn sum(&self) -> PyPropertyFilterOps;
-    fn avg(&self) -> PyPropertyFilterOps;
-    fn min(&self) -> PyPropertyFilterOps;
-    fn max(&self) -> PyPropertyFilterOps;
+    fn len(&self) -> PyResult<PyPropertyFilterOps>;
+
+    fn sum(&self) -> PyResult<PyPropertyFilterOps>;
+
+    fn avg(&self) -> PyResult<PyPropertyFilterOps>;
+
+    fn min(&self) -> PyResult<PyPropertyFilterOps>;
+
+    fn max(&self) -> PyResult<PyPropertyFilterOps>;
+}
+
+trait DynElemQualifierOps: Send + Sync {
+    fn any(&self) -> PyResult<PyPropertyFilterOps>;
+
+    fn all(&self) -> PyResult<PyPropertyFilterOps>;
+}
+
+#[derive(Clone)]
+struct NoElemQualifiers;
+
+impl DynElemQualifierOps for NoElemQualifiers {
+    fn any(&self) -> PyResult<PyPropertyFilterOps> {
+        Err(PyTypeError::new_err("Element qualifiers (any/all) cannot be used after a list aggregation (len/sum/avg/min/max)."))
+    }
+
+    fn all(&self) -> PyResult<PyPropertyFilterOps> {
+        Err(PyTypeError::new_err("Element qualifiers (any/all) cannot be used after a list aggregation (len/sum/avg/min/max)."))
+    }
+}
+
+#[derive(Clone)]
+struct NoListAggOps;
+
+impl DynListAggOps for NoListAggOps {
+    fn len(&self) -> PyResult<PyPropertyFilterOps> {
```

Comment:
We need to have a think about what happens with lists of lists and further nesting